packages/vitest/src/node/cli/cli-api.ts

```diff
@@ -163,6 +173,45 @@ export async function prepareVitest(
   return ctx
 }
 
+export async function collectAndProcess(
+  ctx: Vitest,
+  options: CliOptions,
+  cliFilters: string[],
+) {
+  try {
+    if (!options.filesOnly) {
+      const { tests, errors } = await ctx.collect(cliFilters.map(normalize))
+
+      if (errors.length) {
+        console.error('\nThere were unhandled errors during test collection')
+        errors.forEach(e => console.error(e))
+        console.error('\n\n')
+        await ctx.close()
+        return
+      }
+
+      processCollected(ctx, tests, options)
+    }
+    else {
+      const files = await ctx.listFiles(cliFilters.map(normalize))
+      outputFileList(files, options)
+    }
+
+    await ctx.close()
+  }
+  catch (e) {
```

Comment:
This is interesting part. If we add the additional handling, we need to add it for every error like on line 102. I think how it works right now in `main` is a bug, but it is out of scope of this PR because we also need to add more tests for this. Let's keep it like it was before and track this in another issue